### ISSUE TITLE
[Bug] something wrong when calculating the GLOPs for Linear

### ISSUE THREAD

**Issue by xushilin1** [open-mmlab/mmcv#2936]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmcv/issues) and [Discussions](https://github.com/open-mmlab/mmcv/discussions) but cannot get the expected help.
- [X] The bug has not been fixed in the latest version(https://github.com/open-mmlab/mmcv).

### Environment

MMCV==2.0

### Reproduces the problem - code sample

```python
from mmcv.cnn.utils.flops_counter import add_flops_counting_methods, print_model_with_flops,flops_to_string, params_to_string
import torch

class Net(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.linear = torch.nn.Linear(10, 10)

    def forward(self, x):
        return self.linear(x)

model = Net()
input = torch.randn(32, 10)

flops_model = add_flops_counting_methods(model)
flops_model.eval()
flops_model.start_flops_count()
_ = flops_model(input)
flops_count, params_count = flops_model.compute_average_flops_cost()
print_model_with_flops(flops_model, flops_count, params_count)
flops_model.stop_flops_count()
print(flops_count, params_count)
```

### Reproduces the problem - command or script

None

### Reproduces the problem - error message

GFLOPs = input_batches $\times$ input_channel $\times$ output_channels = $32\times 10\times 10$. But the result of MMCV given is just $10\times10$

**Comment by zhouzaida** [comment]
Hi @xushilin1 , the calculation of FLOPs does not include the batch size dimension. This is because users might use different batch sizes when calculating, which would make comparisons inaccurate.

**Comment by xushilin1** [comment]
Hi, Thanks for your reply!
When I use the FLOP calculation tool in mmengine, I get a different FLOPs result( 3200=32x10x10 compared with 100=10x10 when using mmcv). There may be some wrong in mmcv. 
```python
from mmengine.analysis import get_model_complexity_info
model = Net()
analysis_results = get_model_complexity_info(model, (32,10))
analysis_results['flops_str']
```

**Comment by zhouzaida** [comment]
The input should not contain the **batch size** dim. The correct usage is as follows:

```python
model = Net()
analysis_results = get_model_complexity_info(model, (10,))
analysis_results['flops_str']
```

You can see the details at https://github.com/open-mmlab/mmengine/blob/53474ef1ba0b166508c231fa525b55b580adf20f/mmengine/analysis/print_helper.py#L730
